src/connector/src/schema/schema_registry/client.rs

```diff
@@ -91,7 +95,23 @@ impl Client {
         }
 
         // `unwrap` as the builder is not affected by any input right now
-        let inner = reqwest::Client::builder().build().unwrap();
+        let mut client_builder = reqwest::Client::builder();
+        if let Some(ca_path) = client_config.ca_option.as_ref() {
+            if ca_path.eq_ignore_ascii_case("ignore") {
+                client_builder = client_builder.danger_accept_invalid_certs(true);
+            } else {
+                client_builder = client_builder.add_root_certificate(
+                    reqwest::Certificate::from_pem(&std::fs::read(ca_path).map_err(|e| {
+                        invalid_option_error!("read ca file error: {}", e.as_report())
+                    })?)
+                    .map_err(|e| invalid_option_error!("parse ca file error: {}", e.as_report()))?,
+                );
+            }
+        }
+
+        let inner = client_builder.build().map_err(|e| {
+            invalid_option_error!("build schema registry client error: {}", e.as_report())
```

Comment:
Avoid INTERPOLATION of source error messages (`{}` on `e.as_report()`)! Define a dedicated error to keep the source error:
```
#[derive(thiserror::Error)]
enum ClientBuildError {
  #[error(transparent)]
  InvalidOption(#[from] InvalidOptionError),
  #[error("read ca file error: {0}")]
  ReadFile(#[source] std::io::Error),
  #[error("parse ca file error: {0}")]
  ParsePrm(#[source] reqwest::Error),
  #[error("build schema registry client error: {0}")]
  Build(#[source] reqwest::Error),
}
```
So that:
* `Client::new` returns `ClientBuildError` rather than `InvalidOptionError`
* callers of `Client::new` may eventually transform `ClientBuildError` into a new variant of `SchemaFetchError`

cc @BugenZhao if there are simpler ways